packages/mui-material/src/Slider/Slider.d.ts

```diff
@@ -296,6 +305,53 @@ export declare const SliderTrack: React.FC<SliderTrackProps>;
 export declare const SliderThumb: React.FC<SliderThumbProps>;
 export declare const SliderValueLabel: React.FC<SliderValueLabelProps>;
 
+interface SliderType {
+  <
+    RootComponent extends React.ElementType,
+    Value extends number | number[],
+    OnChangeValue extends Value,
+  >(
+    props: {
+      /**
+       * The component used for the root node.
+       * Either a string to use a HTML element or a component.
+       */
+      component: RootComponent;
+      value?: Value;
```

Comment:
Thinking out loud here: I'm not sure if it could work but have you tried having two overloads instead of generic functions?
```ts
{
    value?: number;
    onChange?: (event: Event, value: number, activeThumb: number) => void;
},
{
    value?: number[],
    onChange?: (event: Event, value: number[], activeThumb: number) => void;
}
```

Reply:
This seems to be working:
```ts
export interface SliderTypeMap<
  RootComponent extends React.ElementType = 'span',
  AdditionalProps = {},
  Value extends number | number[] = number | number[],
> {
  props: AdditionalProps & SliderOwnProps<Value>;
  defaultComponent: RootComponent;
}

export interface SliderType {
  <RootComponent extends React.ElementType>(
    props: {
      component: RootComponent;
    } & OverrideProps<SliderTypeMap<SliderTypeMap['defaultComponent'], {}, number>, RootComponent>,
  ): JSX.Element | null;
  (
    props: DefaultComponentProps<SliderTypeMap<SliderTypeMap['defaultComponent'], {}, number>>,
  ): JSX.Element | null;
  <RootComponent extends React.ElementType>(
    props: {
      component: RootComponent;
    } & OverrideProps<
      SliderTypeMap<SliderTypeMap['defaultComponent'], {}, number[]>,
      RootComponent
    >,
  ): JSX.Element | null;
  (
    props: DefaultComponentProps<SliderTypeMap<SliderTypeMap['defaultComponent'], {}, number[]>>,
  ): JSX.Element | null;
}
```

...but a few tests with undefined `value` fail. Could you explore this direction a bit? I can help you if you hit any roadblocks. Having just one type parameter simplifies the code and (possibly?) allows to remove duplicate prop definitions (as noted in https://github.com/mui/material-ui/pull/38753#discussion_r1315164029)